Source/OCMockTests/OCMockObjectMacroTests.m

```diff
@@ -177,11 +193,24 @@ - (void)testSetsUpStubsWithStructureReturnValues
 
 - (void)testSetsUpStubReturningNilForIdReturnType
 {
-    id mock = OCMClassMock([NSString class]);
+    id mock = OCMPartialMock([NSArray arrayWithObject:@"Foo"]);
+
+    OCMExpect([mock lastObject]).andReturn(nil);
+    XCTAssertNil([mock lastObject], @"Should have returned stubbed value");
+
+    OCMExpect([mock lastObject]).andReturn(Nil);
+    XCTAssertNil([mock lastObject], @"Should have returned stubbed value");
+}
+
+- (void)testSetsUpStubReturningNilForClassReturnType
+{
+    id mock = OCMPartialMock([[TestClassWithClassReturnMethod alloc] init]);
 
-    OCMStub([mock lowercaseString]).andReturn(nil);
+    OCMExpect([mock method]).andReturn(nil);
```

Comment:
Why does the test check `nill` for methods returning `Class`?

Reply:
Same as above in that I think I added it as an extra verification that Nil and nil were equivalent in this case and we could handle both.

Reply:
I see. I guess it isn't too uncommen for folks to use `nil` when it should be `Nil`.